fix: Delete artifact bundle's assemble status when it is deleted

loewenheim opened the pull request:
Assemble statuses persist in the cache for a while, which can result in the artifact bundle being reported to `sentry-cli` as already existing after deletion.

Should fix https://github.com/getsentry/sentry-cli/issues/1724.

Swatinem reviewed (commented):
this also needs a test that ideally reproduces the problem before fixing it.

loewenheim commented on src/sentry/api/endpoints/artifact_bundles.py:
```diff
@@ -177,6 +178,12 @@ def delete(self, request: Request, project) -> Response:
                             # element, so we just take it out and delete it. This deletion will cascade delete every
                             # connected entity.
                             artifact_bundle.delete()
+                            delete_assemble_status(
```
Alright, I moved the status deletion to the artifact bundle `post_delete` hook.

loewenheim commented:
> this also needs a test that ideally reproduces the problem before fixing it.

Yup, I'm thinking about how to do this 🤔

loewenheim commented:
I rebased this so that the first commit just adds a failing test case (actually, the test case itself already existed, I just added a check for the assemble status). The deletion of the assemble status is now slightly more defensive. We only call `delete_assemble_status` if both the organization id and the checksum are present. I'm not sure how they couldn't be, but it does happen in one of the other tests.
